lib/omniauth-dsds/lib/controller_methods.rb

```diff
@@ -20,26 +20,22 @@ def access_token
       end
 
       def fetch_current_user
-        return nil unless access_token
-        build_user
-
-      rescue OAuth2::Error
-        reset_session
-        nil
+        build_user || (reset_session; nil)
       end
 
       def build_user
-        User.build_from strategy_with_access_token(token: access_token).raw_info
+        UserBuilder.new(
+          app_id:     authentication_application_id,
+          app_secret: authentication_application_secret
+        ).build_user access_token
```

Comment:
It could be nil. Carn't think of another reason.
We could do this tho:
```
class UserBuilder
  def initialize(app_id:,app_secret:,token:)
    ... assign variables ...
  end

  def build_user
    return nil unless token
    .....
  end
end
```
which on second thoughts looks cleaner?